File: codeWars/Title_Case/Title_Case.py
# titleCase("a clash of KINGS", "a an the of");

def titleCase(title, minorWords=""):
    words = title.lower().split()
    minors = minorWords.lower().split()
    initial = ""

    for i in words:
        initial = (
            (
                initial + i.capitalize() + " "
                if initial.__len__() == 0
                else initial + i + " "
            )
            if i in minors or initial.__len__() == 0
            else initial + i.capitalize() + " "
        )

    print(initial.strip())
    return initial

titleCase("a clash of KINGS", "a an the of")
titleCase("THE WIND IN THE WILLOWS", "The In")
titleCase("the quick brown fox")

File: codeWars/Title_Case/Title_Case.js
// titleCase("a clash of KINGS", "a an the of");

function titleCase(title, minorWords) {
    const words = title.toLowerCase().split(" ");
    const minor = minorWords.toLowerCase().split(" ");
    let initial = "";

    for (let i = 0; i < words.length; i++) {
        const element = words[i];
        if (!minor.includes(element)) {
            initial = initial + element.replace(element[0], element[0].toUpperCase()) + " ";
        } else {
            initial = initial + element + " ";
        }
    }

    const result = initial.replace(initial[0], initial[0].toUpperCase()).trim();

    // console.log(result);
    return result
}
